shared/devices/add-device/container.js

```diff
@@ -2,23 +2,44 @@
 import * as Container from '../../util/container'
 import * as DevicesGen from '../../actions/devices-gen'
 import * as ProvisionGen from '../../actions/provision-gen'
+import {NavigationActions} from '@react-navigation/core'
 import AddDevice from '.'
+import flags from '../../util/feature-flags'
 import type {RouteProps} from '../../route-tree/render-route'
 
-const mapDispatchToProps = (dispatch, {navigateUp}) => ({
+const mapDispatchToProps = (dispatch, {navigateUp, navigation}) => ({
   onAddComputer: () => {
-    dispatch(navigateUp())
+    if (flags.useNewRouter) {
+      navigation.dispatch(NavigationActions.back())
+    } else {
+      dispatch(navigateUp())
+    }
     dispatch(ProvisionGen.createAddNewDevice({otherDeviceType: 'desktop'}))
   },
   onAddPaperKey: () => {
-    dispatch(navigateUp())
+    if (flags.useNewRouter) {
+      navigation.dispatch(NavigationActions.back())
+      navigation.dispatch(NavigationActions.navigate({routeName: 'devicePaperKey'}))
```

Comment:
i'm a little wary of this pattern.
1. i think we shouldn't expose the raw react-navigation actions to all our componets. lets start w/ routetree and add things we need there.
2. i think we should always push and not call navigate. i think if we call navigate it can go back up the stack if the route is compatible which is actually not what we ever want. in this case it maybe works but in general it does not and doesn't match the ux we want

Reply:
Yeah, I'm not sure what I was thinking here.